pkg/sql/pgwire/pgerrordoc/pgerrordoc.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package pgerrordoc
+
+import (
+	"strings"
+
+	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/errors"
+)
+
+// WithDocsDetails appends a doc URL that contains information for
+// commonly seen error messages.
+func WithDocsDetails(err error) error {
+	if protoRefreshErr := (*roachpb.TransactionRetryWithProtoRefreshError)(nil); errors.As(err, &protoRefreshErr) {
+		var pageAnchor string
+
+		// TransactionRetryWithProtoRefreshError does not actually wrap the error message,
+		// so we're forced to read the message string.
+		for _, match := range []string{
```

Comment:
Maybe you can use `roachpb.TransactionRetryReason_name` or `roachpb.TransactionRetryReason_value` instad of reconstructing a slice of names again and again here?

(there's also `TransactionAbortedReason_value` / `TransactionAbortedReason_name` to play with)

Reply:
I chose this initially because i wasn't sure all anchors existed on the docs page, but i guess if the anchor doesn't exist that's ok!